imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormat.java

```diff
@@ -32,7 +35,8 @@
      * @return the image format or null if unknown
      */
     @Nullable
-    ImageFormat determineFormat(byte[] headerBytes, int headerSize);
+    ImageFormat determineFormat(byte[] headerBytes, int headerSize, InputStream is)
```

Comment:
Yeah, I think that something like this would probably work better. Trying to find the actual SVG tag would b

If you know beforehand which images will be SVG, another option would be to just set the SVG decoder as an override that decodes all images and don't do a header check. Or add an image format override (e.g. by adding it to ImageDecodeOptions). Then you could just assume the format for a given image without needing to check the header (and just fail if it's invalid). I'm not sure how hard this would be to implement though.